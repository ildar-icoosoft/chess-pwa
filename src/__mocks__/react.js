import React from "react";

React.useEffect = jest.fn();

module.exports = React;
