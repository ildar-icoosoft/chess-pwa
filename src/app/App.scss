.container-fluid {
  width: 1140px;
}
