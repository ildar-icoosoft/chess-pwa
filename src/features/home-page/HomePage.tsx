import React, { FC } from "react";
import { Button, Modal } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import OngoingGamesContainer from "../ongoing-games/OngoingGamesContainer";
import ChallengeAiFormContainer from "../challenge-ai-modal/ChallengeAiFormContainer";

import { RootState } from "../../app/rootReducer";
import {
  hideChallengeAiModal,
  showChallengeAiModal,
} from "../challenge-ai-modal/challengeAiModalSlice";
import { AppDispatch } from "../../app/store";

const HomePage: FC<unknown> = () => {
  const dispatch = useDispatch<AppDispatch>();

  const { isChallengeAiModalVisible } = useSelector(
    (state: RootState) => state.challengeAiModal
  );

  return (
    <>
      <Modal
        show={isChallengeAiModalVisible}
        onHide={() => dispatch(hideChallengeAiModal())}
        animation={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>Play with the computer</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <ChallengeAiFormContainer />
        </Modal.Body>
      </Modal>
      <Button
        variant="primary"
        onClick={() => dispatch(showChallengeAiModal())}
      >
        Play with the computer
      </Button>
      <OngoingGamesContainer />
    </>
  );
};

export default HomePage;
