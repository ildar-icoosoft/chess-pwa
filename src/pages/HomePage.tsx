import React, { FC } from "react";
import OngoingGamesContainer from "../containers/OngoingGamesContainer";
import { Button, Modal } from "react-bootstrap";
import { ChallengeAiFormContainer } from "../containers/ChallengeAiFormContainer";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../app/rootReducer";
import {
  hideChallengeAiModal,
  showChallengeAiModal,
} from "../redux/slices/challengeAiModalSlice";
import { AppDispatch } from "../app/store";

const HomePage: FC<unknown> = () => {
  const dispatch = useDispatch<AppDispatch>();

  const { isChallengeAiModalVisible } = useSelector(
    (state: RootState) => state.challengeAiModal
  );

  return (
    <>
      <Modal
        show={isChallengeAiModalVisible}
        onHide={() => dispatch(hideChallengeAiModal())}
        animation={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>Play with the computer</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <ChallengeAiFormContainer />
        </Modal.Body>
      </Modal>
      <Button
        variant="primary"
        onClick={() => dispatch(showChallengeAiModal())}
      >
        Play with the computer
      </Button>
      <OngoingGamesContainer />
    </>
  );
};

export default HomePage;
