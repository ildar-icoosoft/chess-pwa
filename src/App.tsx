import React, { FC, useCallback, useState } from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import HomePage from "./pages/HomePage";
import GamePage from "./pages/GamePage";
import { LoginForm } from "./components/LoginForm";
import { Button, Modal } from "react-bootstrap";

const App: FC = () => {
  const [modalIsVisible, setModalVisibility] = useState(false);

  const showModal = useCallback(() => {
    setModalVisibility(true);
  }, []);
  const hideModal = useCallback(() => {
    setModalVisibility(false);
  }, []);

  return (
    <Router>
      <Button variant="primary" onClick={showModal}>
        Login / Register
      </Button>
      <Modal show={modalIsVisible} onHide={hideModal} animation={false}>
        <Modal.Header closeButton>
          <Modal.Title>Login</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <LoginForm />
        </Modal.Body>
      </Modal>
      <div>
        <nav>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
          </ul>
        </nav>

        <Switch>
          <Route path="/game/:id">
            <GamePage />
          </Route>
          <Route path="/">
            <HomePage />
          </Route>
        </Switch>
      </div>
    </Router>
  );
};

export default App;
