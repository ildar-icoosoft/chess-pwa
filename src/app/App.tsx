import React, { FC, useEffect } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { useDispatch } from "react-redux";
import "bootstrap/dist/css/bootstrap.min.css";
import HomePage from "../features/home-page/HomePage";
import GamePage from "../pages/GamePage";
import { fetchCurrentUser } from "../features/current-user/currentUserSlice";
import { watchGames } from "../features/data-subscription/dataSubscriptionSlice";
import { startGameClock } from "../features/game-clock/gameClockSlice";
import HeaderContainer from "../features/header/HeaderContainer";
import AuthModalContainer from "../features/auth-modal/AuthModalContainer";

const App: FC = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCurrentUser());
    dispatch(watchGames());
    dispatch(startGameClock());
  }, [dispatch]);

  return (
    <div className="container">
      <Router>
        <HeaderContainer />
        <AuthModalContainer />

        <Switch>
          <Route path="/game/:id">
            <GamePage />
          </Route>
          <Route path="/">
            <HomePage />
          </Route>
        </Switch>
      </Router>
    </div>
  );
};

export default App;
