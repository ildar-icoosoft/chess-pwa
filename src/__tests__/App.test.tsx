import React from "react";
import TestRenderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { App } from "../App";
import mountTest from "../tests/mountTest";
import HomePage from "../pages/HomePage";

jest.useFakeTimers();

jest.mock("../services/api");

describe("App", () => {
  mountTest(App);

  it("Snapshot", () => {
    const tree = TestRenderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("children components", () => {
    it("contains HomePage", () => {
      const testRenderer = TestRenderer.create(<App />);
      const testInstance = testRenderer.root;

      expect(testInstance.findAllByType(HomePage).length).toBe(1);
    });
  });

  // @todo. add tests for App contains HomePage and GamePage
  // @todo. add tests about other content

  describe("DOM structure", () => {
    it("renders learn react link", () => {
      const { getByText } = render(<App />);
      const linkElement = getByText(/Home/i);
      expect(linkElement).toBeInTheDocument();
    });
  });
});
