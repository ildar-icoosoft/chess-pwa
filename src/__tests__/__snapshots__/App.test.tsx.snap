// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
Array [
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Login / Register
  </button>,
  <div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="grid"
    />
  </div>,
]
`;
